feelpp/feel/feeldiscr/minmax.hpp

```diff
@@ -0,0 +1,247 @@
+/**
+ * @file minmax.hpp
+ * @author Christophe Prud'homme <christophe.prudhomme@cemosis.fr>
+ * @brief min and max functions of an element of a function space
+ * @version 0.1
+ * @date 2022-06-21
+ * 
+ * @copyright Copyright (c) 2022 Feel++ Consortium
+ * @copyright Copyright (c) 2022 Université de Strasbourg
+ * 
+ */
+#pragma once
+
+#include <feel/feelalg/glas.hpp>
+#include <feel/feelcore/parameter.hpp>
+#include <feel/feelcore/enumerate.hpp>
+#include <feel/feelcore/serialization.hpp>
+#include <feel/feeldiscr/functionspace.hpp>
+#include <feel/feeldiscr/facet.hpp>
+
+namespace Feel {
+
+/**
+ * @brief compute the min of the element of a function space at DOF points over the range @p r
+ * @ingroup Discretization
+ * 
+ * @tparam Ts variadic template holding the type list of arguments
+ *
+ * @param _range range of element
+ * @param _element element of a function space
+ * @param _op operation to apply to the element
+ *
+ * @code {.cpp}
+ * auto v = Xh->element();
+ * auto [op_v,arg_op_v] = minelt(_range=elements(mesh),_element=v,_op=[](auto const& a, auto const& b){ return a < b; });
+ * @endcode
+ *
+ * @return the value and the dof point coordinates
+ */
+template <typename... Ts>
+auto opelt( Ts&&... v )
+{
+    auto args = NA::make_arguments( std::forward<Ts>( v )... );
+    auto && r = args.get(_range);
+    auto && e = args.get(_element);
+    auto && op = args.get(_op);
+
+    using value_type = typename decay_type<decltype(e)>::value_type;
+    using functionspace_type = typename decay_type<decltype(e)>::functionspace_type;
+    constexpr int nRealDim = functionspace_type::nRealDim;
+    value_type v_min = std::numeric_limits<value_type>::min();
+    value_type v_max = std::numeric_limits<value_type>::max();
+    double op_v{op(v_min,v_max)?v_max:v_min};
+    int op_id{ 0 };
+
+    for ( auto const& rangeElt : r )
+    {
+        auto const& meshElt = boost::unwrap_ref( rangeElt );
+        index_type eid = meshElt.id();
+
+        for ( uint16_type local_id=0; local_id < functionspace_type::fe_type::nLocalDof; ++local_id )
+        {
+            int dofpn = e.functionSpace()->dof()->localToGlobal( eid, local_id, 0 ).index();
+            value_type d =e( dofpn );
+            
+            if ( op( d, op_v ) )
+            {
+                op_v = d;
+                op_id = dofpn;
+            }
+        }
+    }
+    auto [pt,thedof,comp] = e.functionSpace()->dof()->dofPoint(op_id);
```

Comment:
the dofpoint method should be deprecated because the cost in memory  (I try to remove all of this and use the geomap context directly).

Reply:
and what is the replacement then ?

Reply:
we should have a one or two line function that does that

Reply:
you should create a geomap context with POINT contetxt and use it  in the loop just before. Issues with dofpoint are : 
- one call will build all dofpoint
- when the mesh move, the developper should call the rebuild of all dofpoint else we get the node without the moving

Reply:
yes right I forgot about that